src/hotspot/share/opto/loopnode.cpp

```diff
@@ -49,7 +49,10 @@
 // Determine if a node is a counted loop induction variable.
 // NOTE: The method is declared in "node.hpp".
 bool Node::is_cloop_ind_var() const {
-  return (is_Phi() && !as_Phi()->is_copy() &&
+#ifdef ASSERT
+  if (is_Phi()) assert(!as_Phi()->is_copy(), "this phi cannot be a copy");
```

Comment:
This line doesn't seem right because `assert` is empty if NDEBUG is set. 
it will alter the original c++ code.   

better to put predicate in assert like Tobias said. `assert(!is_Phi() || !as_Phi()->is_copy() ...)`